cmd/thanos/rule.go

```diff
@@ -758,3 +738,47 @@ func addDiscoveryGroups(g *run.Group, c *http_util.Client, interval time.Duratio
 		cancel()
 	})
 }
+
+type errInvalidPattern struct {
+	err error
+	pat string
+}
+
+func (e *errInvalidPattern) Error() string {
+	return errors.Wrapf(e.err, "retrieving rule files failed. Ignoring file. pattern %s", e.pat).Error()
+}
+func reloadRules(logger log.Logger,
```

Comment:
This function has too many parameters, it makes it harder to read. And most of the parameters are metrics, consider using a struct to collect metrics as it had done in https://github.com/thanos-io/thanos/blob/021f623e7f3b29121f425ec2ce10e4e96377963a/cmd/thanos/downsample.go#L52-L55

and pass it around.

Reply:
@kakkoyun  Committed. Please help to review. Thanks.

Reply:
Looks good now :+1: